pkg/stub/handler.go

```diff
@@ -2,96 +2,410 @@ package stub
+	err := h.secreteclientwrapper.Delete("openshift", h.registrySecret.Name, &metav1.DeleteOptions{})
+	if err != nil && !kerrors.IsNotFound(err) {
+		return h.processError(v1alpha1.SecretUpdateFailed, err, "failed to delete before create dockerconfig secret the openshift namespace: %v")
+	}
+	if deleted {
+		logrus.Printf("registry dockerconfig secret %s was deleted", h.registrySecret.Name)
+		h.registrySecret = nil
+		h.operatorCfg.Status.SecretCreated = false
+		h.sdkwrapper.Update()
+	} else {
+		secretToCreate := corev1.Secret{}
+		h.registrySecret.DeepCopyInto(&secretToCreate)
+		secretToCreate.Namespace = ""
+		secretToCreate.ResourceVersion = ""
+
+		_, err = h.secreteclientwrapper.Create("openshift", &secretToCreate)
+		if err != nil {
+			return h.processError(v1alpha1.SecretUpdateFailed, err, "failed to create registry dockerconfig secret in openshift namespace : %v")
+		}
+		logrus.Printf("created/updated dockerconfig secret %s in openshift namespace", h.operatorCfg.Spec.RegistryCredentials.Name)
+
+		h.operatorCfg.Status.SecretCreated = true
+		h.sdkwrapper.Update()
+
+	}
+
+	status := v1alpha1.ResourceCondition{
+		Type: v1alpha1.SecretUpdateComplete,
+		Time: kapis.Now(),
+	}
+	h.operatorCfg.Status.Conditions = append(h.operatorCfg.Status.Conditions, status)
+	err = h.sdkwrapper.Update()
+	if err != nil {
+		if !kerrors.IsConflict(err) {
+			return h.processError(v1alpha1.SamplesUpdateFailed, err, "failed adding success condition to config: %v")
+		}
+		logrus.Println("got conflict error on success status update, just going to forgo this time vs. retry")
+	}
+
+	return nil
 }
 
 func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
-	switch o := event.Object.(type) {
+	switch event.Object.(type) {
+	case *corev1.Secret:
+		dockercfgSecret, _ := event.Object.(*corev1.Secret)
+
+		if h.operatorCfg == nil {
+			h.registrySecret = dockercfgSecret
+			return nil
+		}
+
+		if h.operatorCfg.Spec.RegistryCredentials.Name == dockercfgSecret.Name {
+			if h.registrySecret != nil &&
+				h.registrySecret.ResourceVersion == dockercfgSecret.ResourceVersion &&
+				h.operatorCfg.Status.SecretCreated &&
+				!event.Deleted {
+				return nil
+			}
+
+			h.registrySecret = dockercfgSecret
+
+			err := h.manageDockerCfgSecret(event.Deleted)
+			if err != nil {
+				return err
+			}
+		}
+
 	case *v1alpha1.SamplesResource:
-		logrus.Printf("ctrl cfg %#v", o)
-		restconfig, err := getRestConfig()
-		if err != nil {
-			logrus.Errorf("failed to get rest config : %v", err)
-			return err
+		if event.Deleted {
+			h.operatorCfg = nil
+			return nil
 		}
-		tempclient, err := getTemplateClient(restconfig)
-		if err != nil {
-			logrus.Errorf("failed to get template client : %v", err)
-			return err
+
+		opcfg, _ := event.Object.(*v1alpha1.SamplesResource)
+
+		if h.operatorCfg != nil && h.operatorCfg.ResourceVersion == opcfg.ResourceVersion {
+			return nil
+		}
+		h.operatorCfg = opcfg
```

Comment:
since you're caching the resource version so you can "no-op" the next time you see this resource(unchanged), I don't think you can set the cache until you're sure you've successfully taken all necessary action to deal with it.

The same is true for where you're caching the secret.

If you were to cache the secret, then fail to copy the secret into the openshift namespace, you'd never have another chance to do it (until the operator was restarted)

Reply:
changes part of next push

on the secrets, since it is less disruptive, I'm not blocking an repeat update, and only using the field to manage the secret event arriving before the CR event